lib/get-exports.js

```diff
@@ -27,6 +27,72 @@ function getExportsForNodeBuiltIn (name) {
 
 const urlsBeingProcessed = new Set() // Guard against circular imports.
 
+/**
+ * This function looks for the package.json which contains the specifier trying to resolve.
+ * Once the package.json file has been found, we extract the file path from the specifier
+ * @param {string} specifier The specifier that is being search for inside the imports object
+ * @param {URL|string} fromUrl The url from which the search starts from
+ * @returns file to url to file export
+ */
+function resolvePackageImports (specifier, fromUrl) {
+  if (!specifier.startsWith('#')) {
+    return null
+  }
+
+  try {
+    const fromPath = fileURLToPath(fromUrl)
+    let currentDir = dirname(fromPath)
+
+    // search for package.json file which has the real url to export
+    while (currentDir !== dirname(currentDir)) {
+      const packageJsonPath = join(currentDir, 'package.json')
+
+      let packageJson
+      try {
+        packageJson = JSON.parse(readFileSync(packageJsonPath, 'utf8'))
+      } catch (error) {
+        if (error.code === 'ENOENT') {
+          currentDir = dirname(currentDir)
+          continue
+        }
+        throw error
```

Comment:
The error itself will be discarded in the catch.

I think it would be better to not rethrow and instead break the loop and just define a state (error variable). That way this would not end up in the catch part again, which should be faster.